### PR TITLE
Added Document resource API methods

### PR DESCRIPTION
**Pull request by hawschiat**
#### Purpose / goal

Added methods to access Document resource API:
- get zipped documents
- get/update documents visibility
- get/update/delete individual document
- get document in original/pdf format
- get document page
- bulk create/delete documents

#### Key changes

- [x] Added methods for the use cases (see above) to the `DocumentResource` class
- [x] (Unit testing) Switched from snapshots in external files to inlined snapshots for better visibility

## Checklists

- [x] Wrote unit tests
- [x] Wrote API documentation

### CODE REVIEW

**Review by abbieschenk — APPROVED** [review]
✅

File: src/resources/DocumentResource.ts [hunk 1]
```diff
@@ -27,14 +26,228 @@ export class DocumentResource extends Resource {
...
+   * @param packageId - Package ID
+   * @param documents - Metadata and data of the to-be-uploaded documents
+   * @returns Metadata of the uploaded documents
+   *
+   * @remarks
+   * - {@link https://community.onespan.com/products/onespan-sign/sandbox#/Documents/api.packages._packageId.documents.post | REST API documentation (OneSpan)}
+   *
+   * - {@link https://community.onespan.com/documentation/onespan-sign/guides/feature-guides/developer/uploading-deleting-documents | Uploading & Deleting Documents (OneSpan)}
+   */
+  public async bulkCreate(
```
Comment (francisco-jara) [inline src/resources/DocumentResource.ts:51]: Very nice! This will definitely come in handy!

File: src/resources/DocumentResource.ts [hunk 2]
```diff
@@ -27,14 +26,228 @@ export class DocumentResource extends Resource {
...
+      .get(`/api/packages${packageId}/documents/${documentId}/pages/${pageIndex}`)
+      .withAcceptHeader('image/png, application/json')
+      .fetch();
+  }
+
+  /**
+   * Retrieves all documents of a package in a zipped file.
+   *
+   * @param packageId - Package ID
+   * @param flatten - (Optional) Flattens the PDFs
```
Comment (francisco-jara) [inline src/resources/DocumentResource.ts:139]: Nice! Didn't know OneSpan supported this through the API.

File: src/types/models/document.ts [hunk 3]
```diff
@@ -146,3 +146,47 @@ export interface DocumentMetadata {
...
+ *
+ * @remarks
+ * Documents not present in the configurations are visible to all recipients. The package owner
+ * can always see all documents in the package, regardless of its document visibility configuration.
+ *
+ * See: {@link https://community.onespan.com/products/onespan-sign/sandbox#/Documents/api.packages._packageId.documents.visibility.post | REST API documentation (OneSpan)}
+ *
+ * @public
+ */
+export interface DocumentVisibility {
```
Comment (francisco-jara) [inline src/types/models/document.ts:182]: Screen Shot 2022-03-04 at 2 31 48 PM

**Review by francisco-jara — APPROVED** [review]
Not sure how you do it--this is a ton of good work 🙇

You are already supporting features we don't have in the current client... we should talk soon about creating an epic to migrate ESign to this lib.

**Comment by hawschiat** [comment]
> Not sure how you do it--this is a ton of good work 🙇

😄 

> You are already supporting features we don't have in the current client... we should talk soon about creating an epic to migrate ESign to this lib.

Yup, we can have a feature branch to field test the library first

**Pull request merged**